artiq/coredevice/phaser.py

```diff
@@ -1039,6 +1060,39 @@ def en_trf_out(self, rf=1, lo=0):
             data = data ^ ((1 << 12) | (1 << 13))
         self.trf_write(data)
 
+    @kernel
+    def set_servo(self, profile=0, bypass=1, hold=0):
+        """Set the servo configuration.
+
+        :param bypass: 1 to enable bypass (default), 0 to engage servo
+        :param hold: 1 to hold the servo IIR filter output constant, 0 for normal operation
+        :param profile: profile index to select for channel (0 to 3)
+        """
+        if (profile < 0) or (profile > 3):
+            raise ValueError("invalid profile index")
+        addr = PHASER_ADDR_SERVO_CFG0 + self.index
+        data = (profile << 2) | ((hold & 1) << 1) | (~bypass & 1)
```

Comment:
Are you forcing hold on bypass in gateware now?

Reply:
No. Should I do it in gateware? Or force it here in SW?

Reply:
I was also thinking that it might actually be beneficial to have a hold option per profile. If I want to save an output amplitude with hold and then do dynamic control on another profile, I have to remember to first enable hold for the IIR and then switch profile, or I loose the "saved" output amplitude. What do you think?

Reply:
I am now forcing hold in the driver method.